plugins/modules/aws_eks_fargate_profile.py

```diff
@@ -0,0 +1,332 @@
+    name = module.params.get('name')
+    cluster_name = module.params['cluster_name']
+    existing = get_fargate_profile(client, module, name, cluster_name)
+    wait = module.params.get('wait')
+    if not existing:
+        module.exit_json(changed=False)
+    if not module.check_mode:
+        check_profiles_status(client, module, cluster_name)
+        try:
+            client.delete_fargate_profile(clusterName=cluster_name, fargateProfileName=name)
+        except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+            module.fail_json_aws(e, msg="Couldn't delete fargate profile %s" % name)
+
+    if wait:
+        wait_until(client, module, 'fargate_profile_deleted', name, cluster_name)
+
+    module.exit_json(changed=True)
+
+
+def get_fargate_profile(client, module, name, cluster_name):
+    try:
+        return client.describe_fargate_profile(clusterName=cluster_name, fargateProfileName=name)['fargateProfile']
+    except is_boto3_error_code('ResourceNotFoundException'):
+        return None
+
+
+# Check if any fargate profiles is in changing states, if so, wait for the end
+def check_profiles_status(client, module, cluster_name):
+    try:
+        list_profiles = client.list_fargate_profiles(clusterName=cluster_name)
+
+        for name in list_profiles["fargateProfileNames"]:
+            fargate_profile = get_fargate_profile(client, module, name, cluster_name)
+            if fargate_profile["status"] == 'CREATING':
+                wait_until(client, module, 'fargate_profile_active', fargate_profile["fargateProfileName"], cluster_name)
+            elif fargate_profile["status"] == 'DELETING':
+                wait_until(client, module, 'fargate_profile_deleted', fargate_profile["fargateProfileName"], cluster_name)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Couldn't not find EKS cluster")
+
+
+def wait_until(client, module, waiter_name, name, cluster_name):
+    wait_timeout = module.params.get('wait_timeout')
+    waiter = get_waiter(client, waiter_name)
+    attempts = 1 + int(wait_timeout / waiter.config.delay)
+    waiter.wait(clusterName=cluster_name, fargateProfileName=name, WaiterConfig={'MaxAttempts': attempts})
+
+
+def main():
+    argument_spec = dict(
+        name=dict(required=True),
+        cluster_name=dict(required=True),
+        role_arn=dict(),
+        subnets=dict(type='list', elements='str'),
+        selectors=dict(type='list', elements='dict'),
+        tags=dict(type='dict', default={}),
+        purge_tags=dict(type='bool', default=True),
+        state=dict(choices=['absent', 'present'], default='present'),
+        wait=dict(default=False, type='bool'),
+        wait_timeout=dict(default=1200, type='int')
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        required_if=[['state', 'present', ['cluster_name', 'role_arn', 'subnets', 'selectors']]],
```

Comment:
some words about that in the documentation would be awesome.

Reply:
I improved the documentation at this point.